File: server/src/main/java/com/example/schoolsite/services/EmployeeService.java
package com.example.schoolsite.services;

import com.example.schoolsite.entity.Teacher;

import java.util.ArrayList;
import java.util.List;

public interface EmployeeService {
    List<Teacher> getAllTeacher();

    List<Teacher> getTeacherByFIO(Teacher teacher);

    Teacher getTeacherByUserId(String userId);
}

File: server/src/main/java/com/example/schoolsite/services/EmployeeService.kt
package com.example.schoolsite.services

import com.example.schoolsite.entity.Teacher

interface EmployeeService {
    val allTeacher: List<Any?>?
    fun getTeacherByFIO(teacher: Teacher?): List<Teacher?>?
    fun getTeacherByUserId(userId: String?): Teacher?
}